tests/main/test_usage_dict.py

```diff
@@ -0,0 +1,63 @@
+from langroid.agent.base import Agent, AgentConfig
+from langroid.cachedb.redis_cachedb import RedisCacheConfig
+from langroid.language_models.base import StreamingIfAllowed
+from langroid.language_models.openai_gpt import OpenAIChatModel, OpenAIGPTConfig
+from langroid.parsing.parser import ParsingConfig
+from langroid.prompts.prompts_config import PromptsConfig
+from langroid.utils.configuration import Settings, set_global
+from langroid.vector_store.base import VectorStoreConfig
+
+
+class CustomAgentConfig(AgentConfig):
+    max_tokens: int = 10000
+    vecdb: VectorStoreConfig = None
+    llm: OpenAIGPTConfig = OpenAIGPTConfig(
+        type="openai",
+        chat_model=OpenAIChatModel.GPT3_5_TURBO,
+        use_chat_for_completion=True,
+        cache_config=RedisCacheConfig(fake=False),
+    )
+    parsing: ParsingConfig = ParsingConfig()
+
+    prompts: PromptsConfig = PromptsConfig(
+        max_tokens=1000,
+    )
+
+
+def test_usage_dict_cache_true():
```

Comment:
1. These tests should be combined into one, using pytest.mark.parameterize, 
to test cache=true, false. 
2. Be aware that when doing `with StreamingIfAllowed(..., False)`, we are actually disabling streaming. The effect of `with StreamingIfALlowed(..., b)` is to allow streaming iff globally allowed AND b = True.
3. The test is focusing on caching, but a more meaningful test is to verify costs and usage tokens are being computed correctly, e.g. compute cost, tokens manually and verify they match the values in the `usage_dict.`
4. `ChatAgent` is much more commonly used than `Agent`, so the tests should include mult-round `ChatAgent` conversation and check that costs/usage calcs are correct.

Reply:
> 3\. e.g. compute cost, tokens manually

can you pls elaborate on `manually`, I noticed the `num_token` (which is used to compute # of `prompt` and `completion` tokens) is already tested manually in various test cases such as `test_code_parser.py`.

Reply:
The general point is that you're accumulating counts into `LLMTokenUsage`, and the test should check that it is doing it correctly, in a multi-round conversation. 
note that `test_code_parser.py` is not really checking the validity of `num_tokens`